app/scripts/components/viewer3dcomponent.js

```diff
@@ -82,12 +82,13 @@ var exports = (function () {
         "deep": true,
         "handler": function () {
           let suid = _.get(this.activeModel, "data.suid");
+          let progress = _.get(this.activeModel, "progress", 0);
           let timestep = _.get(this.activeModel, "data.info.delta_fringe_images.images", []).length;
           let sedimentClass = _.get(this.activeModel, "data.parameters.composition.value");
 
           if (suid !== this.curSuid && timestep !== 0 && sedimentClass !== undefined) {
             this.curSuid = suid;
-            this.curFrameLength = this.curTimeStep = timestep;
```

Comment:
It appears you are quite right. Thanks for this. I will update the code and recommit.

The `timeStep` and `this.curTimeStep` naming convention (as apposed to `timeIndex`) is based on the 3D Viewer API definition: `viewer3d.volume.setTimeStep()` (though I will convert `timestep` to `timeStep`).